packages/sanity/src/structure/panes/document/documentPanel/banners/DeletedDocumentBanners.tsx

```diff
@@ -0,0 +1,109 @@
+import {ReadOnlyIcon} from '@sanity/icons'
+import {Text} from '@sanity/ui'
+import {useCallback, useEffect, useState} from 'react'
+import {
+  type BundleDocument,
+  LATEST,
+  Translate,
+  useBundles,
+  useDocumentOperation,
+  usePerspective,
+  useTimelineSelector,
+  useTranslation,
+} from 'sanity'
+import {useRouter} from 'sanity/router'
+
+import {structureLocaleNamespace} from '../../../../i18n'
+import {useDocumentPane} from '../../useDocumentPane'
+import {Banner} from './Banner'
+
+const useIsLocaleBundleDeleted = () => {
+  const {currentGlobalBundle} = usePerspective()
+  const {data: bundles, deletedBundles} = useBundles()
+  const {slug: currentGlobalBundleSlug} = currentGlobalBundle
+  const [checkedOutBundleSlug, setCheckedOutBundleSlug] = useState<string | undefined>(
+    currentGlobalBundleSlug,
+  )
+
+  useEffect(() => {
+    if (currentGlobalBundleSlug !== LATEST.slug) {
```

Comment:
The checkedout bundle slug is the version. Latest ie. draft slug, is not considered checkedout, since it's the default and fallback, so the state value never reflects drafts. This allows for support in cases where `usePerspective` returns `drafts` quicker than `deletedBundles` updates, and in this case the remote deleted bundle would never be tracked and the banner would never be displayed

Reply:
Great point, I think that adding a similar comment on the code itself would do us good for the future so we can keep this context :)

Reply:
- [ ] Comment on the esoteric logic in `DeletedDocumentBanners`